### PR TITLE
New behavior for ext power control

### PR DESCRIPTION
**Pull request by megamind4089**
Added behaviors to control the external power output.

Tested with nRFMicro 1.3 and Lily58

Behaviors added:
&ext_power EP_ON
&ext_power EP_OFF
&ext_power EP_TOG

I am open to renaming things if there are better suggestions.

A big shoutout to @Tsquash , who helped me to test the behavior with nice!nano and nice!60.

### CODE REVIEW

**Comment by innovaker** [comment]
Thanks @megamind4089!

A couple of Qs on this for @petejohanson:
- Should this be achieved with different behavior tags rather than using a flag?
- Can the shield enhancements be a separate commit? (same PR)

**Comment by petejohanson** [comment]
> Thanks @megamind4089!
> 
> A couple of Qs on this for @petejohanson:
> - Should this be achieved with different behavior tags rather than using a flag?
> - Can the shield enhancements be a separate commit? (same PR)

Using the flag like is there is fine.

I'm not convinced we need the two parameters though. I would think for auto we would just pass `EP_AUTO` and that's that. Leaving as single parameter allows for using this with the hold-tap meta behavior.

**Comment by innovaker** [comment]
Thanks!

I was wondering how we should differentiate between behavior tags and switches (flags) across all behaviors?  What convention should we encourage?  Together, they effectively mean "*do this, in this way*".  It could also be achieved with two tags belonging to the same behavior, effectively `&on` and `&off` - which also optimizes the keymap.

**Comment by megamind4089** [comment]
@innovaker I just used Lily58 to test this behavior and updated with this. I am happy to raise a new PR for shields also.

@petejohanson I have removed the second parameter to the behavior.
Also added toggle behavior as discussed

**Review by petejohanson — CHANGES_REQUESTED** [review]
This is a great start on this. A few tweaks needed.

I think in general, using the "command parameter" like this is good, it avoids "driver instance proliferation", and leverages the argument infrastructure reasonably.

File: app/src/behaviors/behavior_ext_power.c [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+#include <drivers/ext_power.h>
+
+static int on_keymap_binding_pressed(struct device *dev, u32_t position, u32_t command, u32_t arg) {
+    const struct device *ext_power = device_get_binding("EXT_POWER");
+    if (ext_power == NULL) {
+        LOG_ERR("Unable to retrieve ext_power device: %d", command);
+        return -EIO;
+    }
+    const struct ext_power_api *ext_power_api = ext_power->driver_api;
```
[the remark this answers is not in the document]
Comment (megamind4089) [reply app/src/behaviors/behavior_ext_power.c]: What a shame I did. :cry:

File: app/dts/bindings/behaviors/zmk,behavior-ext-power.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright (c) 2020, The ZMK Contributors
+# SPDX-License-Identifier: MIT
+#
+
+description: External power control Behavior
+
+compatible: "zmk,behavior-ext-power"
+
+include: two_param.yaml
```
Comment (petejohanson) [inline app/dts/bindings/behaviors/zmk,behavior-ext-power.yaml:10]: Should be:

```suggestion
include: one_param.yaml
```

File: app/dts/behaviors/ext_power.dtsi [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/ {
+	behaviors {
+		ext_power: behavior_ext_power {
+			compatible = "zmk,behavior-ext-power";
+			label = "EXT_POWER_BEHAVIOR";
+			#binding-cells = <2>;
```
Comment (petejohanson) [inline app/dts/behaviors/ext_power.dtsi:6]: Doesn't need two:

```suggestion
			#binding-cells = <1>;
```

**Review by petejohanson — COMMENTED** [review]
Couple minor tweaks.

File: app/src/behaviors/behavior_ext_power.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (petejohanson) [reply app/src/behaviors/behavior_ext_power.c]: All good!

**Review by petejohanson — APPROVED** [review]
@megamind4089 This is great! Can you please work on documenting the new behavior in a follow up?

Thanks again.

**Pull request merged**